Add neuronexus allego recording Extractor

zm711 opened the pull request:
This depends on:

https://github.com/NeuralEnsemble/python-neo/pull/1509

I still need to upload the test data to GIN and cleanup the PR, but I figure we can have for once that is merged :)

* [x] Add tests

h-mayorquin commented on src/spikeinterface/extractors/neoextractors/neuronexus.py:
```diff
@@ -0,0 +1,66 @@
+            **neo_kwargs,
+        )
+
+        self._kwargs.update(dict(file_path=str(Path(file_path).absolute())))
```
For path resolution I think it is better to use resolve which is like absolute but better.

zm711 replied:
I trust you. I was actually just copying the intan extractor for this line since I'm not pathlib expert. Happy to change :) (and learn more).

zm711 commented on src/spikeinterface/extractors/neoextractors/neuronexus.py:
```diff
@@ -0,0 +1,66 @@
+
+    def __init__(
+        self,
+        file_path,
```
type hints while Sam is away. haha.

h-mayorquin replied:
Oh, I thought that he was against doing them himself I was not aware he dislikes them in principle : O

samuelgarcia replied:
now you know :)

h-mayorquin replied:
x D

Ah, OK, disregard

zm711 commented:
Reminder to myself: we changed it to NeuroNexus at Neo level so I need to fix this for the new name.

zm711 commented on src/spikeinterface/extractors/neoextractors/__init__.py:
```diff
@@ -9,6 +9,7 @@
 from .mearec import MEArecRecordingExtractor, MEArecSortingExtractor, read_mearec
 from .mcsraw import MCSRawRecordingExtractor, read_mcsraw
 from .neuralynx import NeuralynxRecordingExtractor, NeuralynxSortingExtractor, read_neuralynx, read_neuralynx_sorting
+from .neuronexus import NeuronexusRecordingExtractor, read_neuronexus
```
```suggestion
from .neuronexus import NeuronNxusRecordingExtractor, read_neuronexus
```

zm711 commented on src/spikeinterface/extractors/neoextractors/__init__.py:
```diff
@@ -54,6 +55,7 @@
     MCSRawRecordingExtractor,
     NeuralynxRecordingExtractor,
     NeuroScopeRecordingExtractor,
+    NeuronexusRecordingExtractor,
```
```suggestion
    NeuroNexusRecordingExtractor,
```

zm711 commented on src/spikeinterface/extractors/tests/test_neoextractors.py:
```diff
@@ -181,6 +181,14 @@ class NeuroScopeSortingTest(SortingCommonTestSuite, unittest.TestCase):
     ]
 
 
+class NeuronexusRecordingTest(RecordingCommonTestSuite, unittest.TestCase):
```
```suggestion
class NeuroNexusRecordingTest(RecordingCommonTestSuite, unittest.TestCase):
```

zm711 reviewed (commented):
Fix names soon.

zm711 commented on src/spikeinterface/extractors/neoextractors/__init__.py:
```diff
@@ -9,6 +9,7 @@
 from .mearec import MEArecRecordingExtractor, MEArecSortingExtractor, read_mearec
 from .mcsraw import MCSRawRecordingExtractor, read_mcsraw
 from .neuralynx import NeuralynxRecordingExtractor, NeuralynxSortingExtractor, read_neuralynx, read_neuralynx_sorting
+from .neuronexus import NeuronNexusRecordingExtractor, read_neuronexus
```
```suggestion
from .neuronexus import NeuroNexusRecordingExtractor, read_neuronexus
```

samuelgarcia commented:
C'est parfait.
with we have the wraning of importing profiles but I guess we can merge this.

zm711 commented:
Fixed with a rerun. @h-mayorquin import windows failed with 3.05 seconds which was above the limit of 3 second set. Fixed on a re-run. I think we will have to watch out as you mess with numba and cuda--> those lead to really slow imports if not done carefully. Heberto and I will have to go back and fix things later :P

h-mayorquin commented on src/spikeinterface/extractors/neoextractors/neuronexus.py:
```diff
@@ -0,0 +1,66 @@
+
+    Parameters
+    ----------
+    file_path : str
```
This can take path as well, right?

zm711 replied:
Yes because we cast it to a path to resolve. I can add that :)

h-mayorquin commented on src/spikeinterface/extractors/neoextractors/neuronexus.py:
```diff
@@ -0,0 +1,66 @@
+        If there are several streams, specify the stream name you want to load.
+    all_annotations : bool, default: False
+        Load exhaustively all annotations from neo.
+    use_names_as_ids : bool, default: False
```
Can we provide an illustrative examples of the names or are they too variable so giving a pattern does not make sense?

zm711 replied:
I think we don't know the pattern super well. We only have the one test file at Neo, so I could write something, but I'm not confident.

zm711 replied:
Let me know if you think it is still worth putting and I can combine your two pieces of feedback.

h-mayorquin replied:
Na, let's do it when we know more. No point in adding possible confusing message.

h-mayorquin commented on src/spikeinterface/extractors/tests/common_tests.py:
```diff
@@ -52,8 +52,11 @@ def test_open(self):
                 full_traces = rec.get_traces(segment_index=segment_index)
-                assert full_traces.shape == (num_samples, num_chans)
-                assert full_traces.dtype == dtype
+                assert full_traces.shape == (
```
Something I like to do is to write expected_shape = (num_samples, num_channels) and then do the assertion so if I Jump in the debugger I have all the variables already loaded. But I think this is OK.

zm711 replied:
Yeah I didn't change anything here. I was just getting an error and couldn't see what it was so I rage added this :)

h-mayorquin replied:
Yes, I realize, just sharing a pattern. I was going to suggest adding string to the message but this is the testing suite so I think this is fine and your addition is an improvement.

h-mayorquin reviewed (commented):
This LGTM, some simple questions.

zm711 commented on src/spikeinterface/extractors/neoextractors/neuronexus.py:
```diff
@@ -0,0 +1,66 @@
+
+    Parameters
+    ----------
+    file_path : str
```
```suggestion
    file_path : str | Path
```
